### ISSUE TITLE
L059: Allow prefered quoted for identifiers

### ISSUE THREAD

**Issue by niconoe-** [sqlfluff/sqlfluff#2527]
The current rule L059 is raising an error when some unnecessary quoted identifier has been detected.

Although it is not recommended to quote identifiers when it's not needed, this is a convention I am using for automatic business script that are parsing my SQLs. Therefore, I would like to use the backtick as a mandatory wrapper for all identifiers, no matter if they need to be quoted.

I could ignore the rule L059, but I would prefer to let the user allow, through the configuration, which quote to use and have a configuration to actually force it everywhere.

I propose 2 configurations for the L059 rule:
- force_quote_identifier: [True,False] (Defaults: False) # Force every identifiers to be quoted, even when unnecessary if True.
- preferred_quote_identifier: (any character) (Default: ") # Preferred quote identifier to use.

#### Anti-pattern
```sql
SELECT `foo`, "bar", `sum`
FROM table_a;
```
#### Pattern
```sql
SELECT foo, bar, "sum"
FROM table_a;
```

#### sqlfluff.cfg
```cfg
[sqlfluff:rules:L059]
force_quote_identifier: True
preferred_quote_identifier: `
```
#### Anti-pattern
```sql
SELECT foo, "bar", "sum"
FROM table_a;
```
#### Pattern
```sql
SELECT `foo`, `bar`, `sum`
FROM table_a;
```

This doesn't break the current behavior of the L059 rule, but allows a better user-configuration for this rule.

What do you think about?

Thanks :+1:

**Comment by niconoe-** [comment]
Last time I spoke python was almost 10 years ago, so I'll definitly needs a refresh to start working at this, although I could try to propose a PR if I can find some time (this is not top priority as, like I said, I can exclude L059 until this feature is available).

The more I think about mixed quotes (yeah, definitly I pray for no one using this...), the more I assume that if there's an occurrence of the quote identifier inside the identifier name, we should ignore this rule on this specific case, for now. 

I saw an issue talking about criticity of errors, and maybe having levels like "error", "warning" and "info" could help us in setting this kind of (awesomly dirty) mixed quotes use cases in "info" level, like `L059: Declared preferred quote identifier is "…" but this character is used in identifier named "…".` But, to me, this is going very far for a second thought, and probably considering this very specific use case should be ignored first is a good approach.

**Comment by niconoe-** [comment]
Thank you very much for the help you provided to me during this work! Was a pleasure!
